src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            updateDrivers(car, connection);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update cat from carsDB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could`t delete car from carsDB", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.id AS cars_id, "
+                + "cars.model AS model, "
+                + "manufacturers.id AS manufacturers_id, "
+                + "manufacturers.name AS manufacturers_name, "
+                + "manufacturers.country AS manufacturers_country "
+                + "FROM cars "
+                + "JOIN cars_drivers "
+                + "ON cars.id = cars_drivers.car_id "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + " WHERE driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            List<Car> cars = new ArrayList<>();
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                car.setDrivers(getDrivers(car.getId(), connection));
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars by driver", e);
+        }
+    }
+
+    private List<Driver> getDrivers(Long carId, Connection connection) {
+        String query = "SELECT * FROM drivers "
+                + "JOIN cars_drivers "
+                + "ON cars_drivers.driver_id = drivers.id "
+                + "WHERE cars_drivers.car_id = ?;";
```

Comment:
```suggestion
                + "WHERE cars_drivers.car_id = ? AND cars.is_deleted = false AND drivers.is_deleted = false;";
```
here you also should filter deleted cars